tests/BuildStrategy.cxx

```diff
@@ -0,0 +1,62 @@
+#include <FastBVH.h>
+
+#include "Primitive.h"
+
+#include <gtest/gtest.h>
+
+#include <list>
+
+using namespace FastBVH;
+
+namespace {
+
+//! \brief This is just boilerplate code for google test
+//! to be able to run template tests.
+template <typename T>
+class BuildStrategyTest : public ::testing::Test {
```

Comment:
Awesome :+1: Tests will be great. We can get this added to the GitHub Actions to run these checks on new PRs.

Reply:
Awesome! I didn't write any tests yet for the existing code, I figured if it works, it works. I'll be proactive writing tests for new code though.